internal/auth/ldap/repository_auth_method_create.go

```diff
@@ -0,0 +1,157 @@
+		}
+		if err := cc.encrypt(ctx, dbWrapper); err != nil {
+			return nil, errors.Wrap(ctx, err, op, errors.WithMsg("failed to encrypt client certificate"))
+		}
+	}
+
+	oplogWrapper, err := r.kms.GetWrapper(ctx, am.ScopeId, kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+	}
+
+	var returnedAuthMethod *AuthMethod
+	_, err = r.writer.DoTx(
+		ctx,
+		db.StdRetryCnt,
+		db.ExpBackoff{},
+		func(r db.Reader, w db.Writer) error {
+			var ()
+			msgs := make([]*oplog.Message, 0, 4)
+			ticket, err := w.GetTicket(ctx, am)
+			if err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get ticket"))
+			}
+
+			returnedAuthMethod = am.clone()
+			var amOplogMsg oplog.Message
+			if err := w.Create(ctx, returnedAuthMethod, db.NewOplogMsg(&amOplogMsg)); err != nil {
+				return err
+			}
+			msgs = append(msgs, &amOplogMsg)
+
+			urlOplogMsgs := make([]*oplog.Message, 0, len(cv.Urls))
+			if err := w.CreateItems(ctx, cv.Urls, db.NewOplogMsgs(&urlOplogMsgs)); err != nil {
+				return err
+			}
+			msgs = append(msgs, urlOplogMsgs...)
+
+			if len(cv.Certs) > 0 {
+				certOplogMsgs := make([]*oplog.Message, 0, len(cv.Certs))
+				if err := w.CreateItems(ctx, cv.Certs, db.NewOplogMsgs(&certOplogMsgs)); err != nil {
+					return err
+				}
+				msgs = append(msgs, certOplogMsgs...)
+			}
+			if cv.UserEntrySearchConf != nil {
+				var ucOplogMsg oplog.Message
+				if err := w.Create(ctx, cv.UserEntrySearchConf, db.NewOplogMsg(&ucOplogMsg)); err != nil {
+					return err
+				}
+				msgs = append(msgs, &ucOplogMsg)
+			}
+			if cv.GroupEntrySearchConf != nil {
+				var gcOplogMsg oplog.Message
+				if err := w.Create(ctx, cv.GroupEntrySearchConf, db.NewOplogMsg(&gcOplogMsg)); err != nil {
+					return err
+				}
+				msgs = append(msgs, &gcOplogMsg)
+			}
+			if cv.ClientCertificate != nil {
+				var ccOplogMsg oplog.Message
+				if err := w.Create(ctx, cv.ClientCertificate, db.NewOplogMsg(&ccOplogMsg)); err != nil {
+					return err
+				}
+				msgs = append(msgs, &ccOplogMsg)
+			}
+			if cv.BindCredential != nil {
+				var bcOplogMsg oplog.Message
+				if err := w.Create(ctx, cv.BindCredential, db.NewOplogMsg(&bcOplogMsg)); err != nil {
+					return err
+				}
+				msgs = append(msgs, &bcOplogMsg)
+			}
+			if err := w.WriteOplogEntryWith(ctx, oplogWrapper, ticket, am.oplog(oplog.OpType_OP_TYPE_CREATE), msgs); err != nil {
+				return errors.Wrap(ctx, err, op, errors.WithMsg("unable to write oplog"))
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	found, err := r.getAuthMethods(ctx, am.GetPublicId(), nil)
+	switch {
+	case err != nil:
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("failed to lookup created auth method %q", am.GetPublicId()))
+	case len(found) != 1:
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("found %q auth method (%q) and expected 1", len(found), am.GetPublicId()))
```

Comment:
`%q` is here used with an integer, but is only supported with strings and byte slices according to https://pkg.go.dev/fmt#hdr-Printing. I think you want `%d`?.

Reply:
ty!